exec/plugin-commands-script-runners/src/run.ts

```diff
@@ -175,7 +175,17 @@ so you may run "pnpm -w run ${scriptName}"`,
   const lifecycleOpts: RunLifecycleHookOptions = {
     depPath: dir,
     extraBinPaths: opts.extraBinPaths,
-    extraEnv: opts.extraEnv,
+    // Follow the behavior of npm by setting 'npm_command' env variable to 'run-script' when running scripts (e.g. pnpm run dev)
+    // and setting it to the command name otherwise (e.g. pnpm test)
+    extraEnv: {
+      ...opts.extraEnv,
+      ...(opts.extraEnv.npm_command && {
+        npm_command:
+                  opts.extraEnv.npm_command === 'run'
+                    ? 'run-script'
+                    : opts.extraEnv.npm_command,
+      }),
+    },
```

Comment:
Why are changes needed here if the npm_command field is set in main.ts?

Reply:
That's a very good question 😺 Thanks for noticing this! Must have been a bit tired... I've now moved it to `main.ts`.